lib/parser/provenance.h

```diff
@@ -143,6 +144,8 @@ class AllSources {
       const std::string &message, bool echoSourceLine = false) const;
   const SourceFile *GetSourceFile(
       Provenance, std::size_t *offset = nullptr) const;
+  std::optional<std::pair<int, int>> GetSourcePosition(Provenance) const;
```

Comment:
I think you should replace the `std::pair` with a struct so that the members have names. It makes it easier to figure out what these two `int`s represent.